### ISSUE TITLE
masked tensor implementation of torch._ops.aten.addmm.default, torch._ops.aten.permute.default, etc.

### ISSUE THREAD

**Issue by MOVzeroOne** [pytorch/pytorch#99727]
### 🚀 The feature, motivation and pitch

I really like the idea behind the new masked tensor, as padding in my situation is very undesirable as I want my network to not rely on certain parts of the input at all. Padding with zeros just doesn't cut it. 
(this is because with zero padding in my specific case the padded region gets used by the network as "extra memory" to store intermediates. This normally results in me having to zero out those regions manually after certain operations) 

When using masked tensor I ran into a problem that torch._ops.aten.addmm.default is not yet implemented and torch._ops.aten.permute.default is are not yet implemented for masked tensor.

Having more operations for masked tensor so it can be used as masked alternative to the default torch.tensor would be amazing. As it will prevent problems caused by the trained network unknowingly relying on the padding.
